File: features/featureTwo/src/main/java/de/gfred/feature/two/di/FeatureTwoActivityModule.java
package de.gfred.feature.two.di;

import de.gfred.shared.models.services.IUserService;
import dagger.Module;
import dagger.Provides;
import de.gfred.feature.two.FeatureTwoPresenter;
import de.gfred.feature.two.IFeatureTwoPresenter;

@Module
public class FeatureTwoActivityModule
{
    @Provides
    public IFeatureTwoPresenter providesFeatureTwoPresenter(final IUserService userService)
    {
        return new FeatureTwoPresenter(userService);
    }
}

File: features/featureTwo/src/main/kotlin/de/gfred/feature/two/di/FeatureTwoActivityModule.kt
package de.gfred.feature.two.di

import dagger.Module
import dagger.Provides
import de.gfred.feature.two.FeatureTwoPresenter
import de.gfred.feature.two.FeatureTwoPresenterImpl
import javax.inject.Singleton

@Singleton
@Module
class FeatureTwoActivityModule {
    @Provides
    fun providesFeatureTwoPresenter(featureTwoPresenterImpl: FeatureTwoPresenterImpl): FeatureTwoPresenter = featureTwoPresenterImpl
}
